src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -107,22 +112,33 @@ async def setup_go_sdk_process(
     request: GoSdkProcess,
     go_sdk_run: GoSdkRunSetup,
     bash: BashBinary,
+    git: GitBinary,
     golang_env_aware: GolangSubsystem.EnvironmentAware,
     goroot: GoRoot,
 ) -> Process:
-    input_digest, env_vars = await MultiGet(
+    binary_shims_get = Get(
```

Comment:
This will require `git` even if `go` does not need it. Maybe the go backend should use a similar approach to the docker backend's `--docker-tools` option which allows the user to specify any number of tools? See https://github.com/pantsbuild/pants/blob/ee2128b31cb8963874d139991b812e14c19dd565/src/python/pants/backend/docker/util_rules/docker_binary.py#L167

Thus there would be a `--golang-subprocess-tools` option (or some better name).

Reply:
@tdyas thanks. I will have a look at that.

Reply:
@tdyas I have changed the golang subprocess to include any binaries specified with a new `--golang-extra-tools` option.
I am also using the `ExecutableSearchPathsOptionMixin` rather than having an additional option for the search path for the extra tools.